receiver/sshcheckreceiver/internal/configssh/configssh.go

```diff
@@ -0,0 +1,163 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configssh // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sshcheckreceiver/internal/configssh"
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/pkg/sftp"
+	"go.opentelemetry.io/collector/component"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/knownhosts"
+)
+
+const (
+	defaultClientVersion = "SSH-2.0-OTelClient"
+)
+
+var errMissingKnownHosts = errors.New(`known_hosts file is missing`)
+
+// TODO add configuration to provide certificate authentication
+
+type SSHClientSettings struct {
+	// Endpoint is always required
+	Endpoint string        `mapstructure:"endpoint"`
+	Timeout  time.Duration `mapstructure:"timeout"`
+
+	// authentication requires a Username and either a Password or KeyFile
+	Username string `mapstructure:"username"`
+	Password string `mapstructure:"password"`
+	KeyFile  string `mapstructure:"key_file"`
+
+	// IgnoreHostKey provides an insecure path to quickstarts and testing
+	IgnoreHostKey bool `mapstructure:"ignore_host_key"`
+}
+
+type Client struct {
+	*ssh.Client
+	*ssh.ClientConfig
+	DialFunc func(network, address string, config *ssh.ClientConfig) (*ssh.Client, error)
+}
+
+// Dial starts an SSH session.
+func (c *Client) Dial(endpoint string) (err error) {
+	c.Client, err = c.DialFunc("tcp", endpoint, c.ClientConfig)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Client) SFTPClient() (*SFTPClient, error) {
+	if c.Client.Conn == nil {
+		return nil, fmt.Errorf("SSH client not initialized")
+	}
+	client, err := sftp.NewClient(c.Client)
+	if err != nil {
+		return nil, err
+	}
+	return &SFTPClient{
+		Client:       client,
+		ClientConfig: c.ClientConfig,
+	}, nil
+}
+
+type SFTPClient struct {
+	*sftp.Client
+	*ssh.ClientConfig
+}
+
+func NewDefaultSSHClientSettings() SSHClientSettings {
```

Comment:
This function is only used in the testing part, and never called from the main code. Does it keep any value?

Reply:
addressed in e9e005a9c